### ISSUE TITLE
[Enhancement] Support for encrypted databags

### ISSUE THREAD

**Issue by ClaesGIS** [Esri/arcgis-cookbook#220]
### Idea
To further improve this cookbook I would like to be able to use encrypted data bags to store sensitive information. To name a few:
- `server['admin_password'] `
- `server['keystore_password']`
-  `portal['security_question']`  & `portal['security_question_answer']`

### Ideas for implementation
- Specify the bag name and key as an attribute
- Specify the location of the shared secret
- Perhaps do something with [environments ](https://docs.chef.io/data_bags.html#with-environments)

Keep in mind I might be thinking about data bags wrong.

**Comment by tsmalls93** [comment]
I agree. The current documentation suggests storing credentials in plain text is okay, but it is actually really insecure.

**Comment by slootsky** [comment]
agreed.  This would be very useful for our environment as well.
